typespec-tests/tsp/multiple-content-types.tsp

```diff
@@ -42,7 +42,11 @@ interface MultipleContentTypesOnRequest {
   @post
   uploadBytesWithSingleBodyTypeForMultiContentTypes(
     @body data: bytes,
-    @header contentType: "application/octet-stream" | "image/jpeg" | "image/png",
+    @header contentType:
+      | "application/octet-stream"
+      | "image/jpeg"
+      | "image/png"
+      | "application/json-patch+json",
```

Comment:
this `application/json-patch+json` triggers the bug.

usual case is good.

Reply:
So the CI failure is due to ‘json-patch+json’ content type, not due to defining multiple content types as enum? I’m a bit confused why defining ‘json-patch+json’ will cause the contentType in method use enum rather than string.

Reply:
https://github.com/Azure/autorest.java/pull/2870#discussion_r1683874251

Reply:
ok, I see.